crdsonnet/renderEngines/ast.libsonnet

```diff
@@ -25,23 +26,36 @@ local xtd = import 'github.com/jsonnet-libs/xtd/main.libsonnet';
   },
 
   nestInParents(name, parents, field)::
-    j.object.members([
+    local startParents =
+      if jutils.isObject(field)
+      then parents[:std.length(parents) - 1]
+      else parents;
+    local startField =
+      if jutils.isField(field) || jutils.isFunction(field)
+      then field
+      else if jutils.isObject(field)
+      then
+        j.field.field(
+          j.fieldname.string(parents[std.length(parents) - 1]),
+          field,
+          additive=true,
+        )
+      else error '`field` is of type %s, must be a field type' % jutils.type(field);
+    [
       std.foldr(
         function(p, acc)
           if p == name
           then acc
           else
-            customField.field(
+            j.field.field(
```

Comment:
This change is not strictly part of this PR, `j.field` improves the static output a bit by adding a few linebreaks instead of taking them away with `customField`.